src/main/java/io/swagger/oas/inflector/controllers/OpenAPIOperationController.java

```diff
@@ -607,14 +665,28 @@ public void setContentType(RequestContext res, ResponseContext resp, Operation o
 
                 for(String key : content.keySet()) {
                     MediaType mediaType = MediaType.valueOf(key);
-                    for (MediaType acceptable : res.getAcceptableMediaTypes()) {
-                        if (mediaType.isCompatible(acceptable)) {
-                            resp.setContentType(mediaType);
-                            return;
+                    if (res.getHeaders().get("Content-Type")!= null) {
+                        for (String acceptable : res.getHeaders().get("Content-Type")) {
+                            String subtype = acceptable.substring(acceptable.lastIndexOf("/") + 1);
+
+                            if (subtype.equals(mediaType.getSubtype())) {
```

Comment:
these statements don't make sense, the expression: `resp.setContentType(mediaType);` is executed no matter if the condition happen.

Is the conditional necessary or the `else` statement should be different?

Reply:
Done. removed the redundant code